### PR TITLE
SOC-1422 | Discussion post and replies editor

### PR DESCRIPTION
**Pull request by rafalkalinski**
The code here is a merged code from two PRs:
https://github.com/Wikia/mercury/pull/1614
https://github.com/Wikia/mercury/pull/1651

Both already got LGTM

https://wikia-inc.atlassian.net/browse/SOC-1422

### CODE REVIEW

File: front/scripts/main/components/discussion-editor.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+	offsetTop: 0,
+	siteHeadHeight: 0,
+
+	bodyText: '',
+	layoutName: 'components/discussion-editor',
+
+	errorMessage: Ember.computed.oneWay('requestErrorMessage'),
+
+	/**
+	 * returns {boolean}
```
Comment (hakubo) [inline front/scripts/main/components/discussion-editor.js]: missing @

File: front/scripts/main/components/discussion-editor.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+	 */
+	submitDisabled: Ember.computed('bodyText', 'currentUser.userId', function () {
+		return this.get('bodyText').length === 0 || this.get('currentUser.userId') === null;
+	}),
+
+	/**
+	 * @returns {void}
+	 */
+	init() {
```
Comment (hakubo) [inline front/scripts/main/components/discussion-editor.js]: consider:
```
init(...params) {
    this._super(...params)
}
```

File: front/scripts/main/components/discussion-editor.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	/**
+	 * Method should be overwritten in the child classes
+	 * @returns {void}
+	 */
+	isStickyBreakpointHeight() {
+		throw new Error('Please, override this method in the descendant class');
+	},
+
+	/**
+	 * Method should be overwritten in the child classes
```
Comment (hakubo) [inline front/scripts/main/components/discussion-editor.js]: missing 
```
@throws 
```

File: front/scripts/main/components/discussion-editor.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+	 * @param {object} newItem
+	 * @returns {void}
+	 */
+	handleNewItemCreated(newItem) {
+		this.setProperties({
+			isLoading: false,
+			showSuccess: true
+		});
+
+		Ember.set(newItem, 'isVisible', false);
```
Comment (hakubo) [inline front/scripts/main/components/discussion-editor.js]: newItem.set('isVisible', false)

File: front/scripts/main/components/discussion-reply-editor.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	placeholderText: 'editor.reply-editor-placeholder-text',
+	submitText: 'editor.reply-action-button-label',
+	labelText: 'editor.reply-editor-label',
+
+	/**
+	 * Initialize onScroll binding for sticky logic
+	 * @returns {void}
+	 */
+	initializeStickyState() {
+		if (window.innerHeight < this.$().offset().top + this.$().height()) {
```
Comment (hakubo) [inline front/scripts/main/components/discussion-reply-editor.js]: this.set('isSticky', window.innerHeight < this.$().offset().top + this.$().height())

File: front/scripts/main/components/discussion-reply-editor.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+			this.initializeStickyState();
+		}
+	),
+
+	/**
+	 * Perform animations and logic after reply creation
+	 * @returns {void}
+	 */
+	handleNewReplyCreated: Ember.observer('replies.@each.isNew', function () {
+		const newReplies = this.get('replies').filter((reply) => {
```
Comment (hakubo) [inline front/scripts/main/components/discussion-reply-editor.js]: `this.get('replies').filter(reply => reply.isNew),

File: front/scripts/main/components/discussion-reply-editor.js (hunk 5 above)
Comment (rafalkalinski) [inline front/scripts/main/components/discussion-reply-editor.js]: I will put it into some new variable.

File: front/scripts/main/components/discussion-editor.js (hunk 3 above)
Comment (rafalkalinski) [inline front/scripts/main/components/discussion-editor.js]: added

**Comment by kvas-damian** [comment]
LGTM

File: front/scripts/main/components/discussion-editor.js (hunk 4 above)
Comment (rafalkalinski) [inline front/scripts/main/components/discussion-editor.js]: It doesn't work here - the context is not an ember model.

**Comment by kvas-damian** [comment]
As discussed offline - we'll continue improvement for iOS in follow up ticket.
For now it LGTM

**Pull request merged**